x/migration/keeper/msg_server_claim_morse_supplier.go

```diff
@@ -187,19 +188,21 @@ func (k msgServer) ClaimMorseSupplier(
 	postClaimSupplierStake := preClaimSupplierStake.Add(morseClaimableAccount.GetSupplierStake())
 	minStake := k.supplierKeeper.GetParams(ctx).MinStake
 
+	currentSessionStart := sharedtypes.GetSessionStartHeight(&sharedParams, sdkCtx.BlockHeight())
+	previousSessionEnd := sharedtypes.GetSessionEndHeight(&sharedParams, currentSessionStart-1)
+
+	// Construct in case it is already or will become unbonded immediately (i.e. below min stake).
```

Comment:
It's a bit confusing we're constructing an unbonded supplier even though we don't need it yet.

I feel like this needs to be restructured or renamed.

Even if we need to make object copies later.

Reply:
There are multiple sibling conditional branches which depend on this variable. I consolidated it into a single instance here to reduce boilerplate.